lib/cli.js

```diff
@@ -385,7 +387,8 @@ const cli = {
         if (useStdin) {
             results = await engine.lintText(text, {
                 filePath: options.stdinFilename,
-                warnIgnored: true
+                // eslint-disable-next-line no-undefined -- flatConfig respects CLI flag and constructor warnIgnored, eslintrc forces true for backwards compatibility
+                warnIgnored: usingFlatConfig ? undefined : true
```

Comment:
```suggestion

                // flatConfig respects CLI flag and constructor warnIgnored, eslintrc forces true for backwards compatibility
                warnIgnored: usingFlatConfig ? void 0 : true
```

We're using `void 0` to get the `undefined` value.